libs/shared/ui/src/lib/components/inputs/input-size-unit/input-size-unit.tsx

```diff
@@ -0,0 +1,81 @@
+import { useState } from 'react'
+import { FieldError, FieldValues, UseFormSetValue } from 'react-hook-form'
+import { MemorySizeEnum } from '@console/shared/enums'
+import { convertMemory } from '@console/shared/utils'
+import InputSelect from '../input-select/input-select'
+import InputText from '../input-text/input-text'
+
+export interface InputSizeUnitProps {
+  name: string
+  value: number
+  currentUnit?: string | MemorySizeEnum
+  maxSize?: number
+  minSize?: number
+  currentSize?: number
+  getUnit?: (value: string | MemorySizeEnum) => void
+  onChange?: () => void
+  error?: FieldError
+  setValue?: UseFormSetValue<FieldValues>
+}
+
+export const getSizeUnit = (memorySize: MemorySizeEnum | string, value: number | string) => {
+  if (memorySize === MemorySizeEnum.GB) {
+    return convertMemory(Number(value), MemorySizeEnum.MB)
+  } else {
+    return convertMemory(Number(value), MemorySizeEnum.GB)
+  }
+}
+
+export function InputSizeUnit(props: InputSizeUnitProps) {
+  const { name, maxSize, currentSize = 0, getUnit, currentUnit, error, value, onChange, setValue } = props
+
+  const [memorySize, setMemorySize] = useState<string | MemorySizeEnum>(currentUnit || MemorySizeEnum.MB)
+
+  const handleChangeMemoryUnit = (size: string | MemorySizeEnum) => {
+    getUnit && getUnit(size)
+    setMemorySize(size)
+
+    if (size !== memorySize) {
+      const currentSizeByUnit = getSizeUnit(size, value)
+      setValue && setValue(name, currentSizeByUnit)
```

Comment:
Remi, I'm not convinced that you need to give the setValue, you can directly do that with the onChange I think.
I mean, you can call onChange({name, currentSizeByUnit})

It makes this component even less complex to use. In the onChange you will have your `field.onChange` that does the same as setValue(name, value)
Do you see what I mean?